File: Backend/model/Property.java
package com.realestate.model;

import java.time.LocalDateTime;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "property")
public class Property {

    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    private String title;

    @Column(columnDefinition = "TEXT")
    private String description;

    private double price;
 
    private String location;

    @Column(name = "image_url", columnDefinition = "TEXT")
    private String imageUrl;

    private boolean approved = false;

    @Column(name = "created_at")
    private LocalDateTime createdAt = LocalDateTime.now();

    @ManyToOne
    @JoinColumn(name = "seller_id")
    private User seller;

    @ManyToOne
    @JoinColumn(name = "type_id")
    private PropertyType type;
    
    @ManyToOne
    @JoinColumn(name = "manager_id")
    private User manager;

}

File: Frontend/admin/model/property.ts

import { User } from './user';
import { PropertyType } from './property-type';

export interface Property {
  id: number;
  title: string;
  description: string;
  price: number;
  location: string;
  imageUrl: string;
  approved: boolean;
  createdAt: string; // ISO date string, e.g. "2025-07-06T12:34:56"
  seller: User;
  type: PropertyType;
  manager: User;
}
